### PR TITLE
replace deploymentconfig with deployment for header-test

### PR DESCRIPTION
**Pull request by lihongan**
to address https://issues.redhat.com/browse/OCPQE-17212

@melvinjoseph86 @ShudiLi PTAL, thanks

log: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Runner/879055/console

### CODE REVIEW

**Comment by ShudiLi** [comment]
/lgtm
Thanks

File: features/step_definitions/helper_services.rb [hunk 1]
```diff
@@ -379,8 +379,8 @@
     raise "project #{project_name} does not exist"
   end
 
-  @result = user.cli_exec(:create, f: "#{BushSlicer::HOME}/testdata/routing/header-test/dc.json")
-  raise "could not create header test dc" unless @result[:success]
+  @result = user.cli_exec(:create, f: "#{BushSlicer::HOME}/testdata/routing/header-test/deployment.yaml")
+  raise "could not create header test deployment" unless @result[:success]
   cb.header_test_dc = dc("header-test")
```
Comment (liangxia) [inline features/step_definitions/helper_services.rb:384]: Should we update this line `dc("header-test")` as well ?

Comment (lihongan) [reply features/step_definitions/helper_services.rb]: yes, updated to deployment

**Comment by lihongan** [comment]
updated the step `Given /^I have a header test service in the#{OPT_QUOTED} project$/` since it also uses header-test deployment.

new logs 
https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Runner/879129/console

**Pull request merged**